sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -283,20 +283,7 @@ case class TakeOrderedAndProjectExec(
   }
 
   override def executeCollect(): Array[InternalRow] = {
-    val orderingSatisfies = SortOrder.orderingSatisfies(child.outputOrdering, sortOrder)
-    val ord = new LazilyGeneratedOrdering(sortOrder, child.output)
-    val limited = if (orderingSatisfies) {
-      child.execute().mapPartitionsInternal(_.map(_.copy()).take(limit)).takeOrdered(limit)(ord)
-    } else {
-      child.execute().mapPartitionsInternal(_.map(_.copy())).takeOrdered(limit)(ord)
```

Comment:
hmm, so `RDD.takeOrdered` does not return ordered records?

Reply:
No. The result does not match if it contains duplicate values. For example, order by column a:

```
+---+---+
|  a|  b|
+---+---+
|  1|  1|
|  1|  2|
|  2|  3|
|  2|  4|
|  3|  5|
|  3|  6|
+---+---+
```